Python/bytecodes.c

```diff
@@ -4941,6 +4941,12 @@ dummy_func(
             DEOPT_IF(func->func_version != func_version);
         }
 
+        tier2 op(_CHECK_FUNCTION_UNMODIFIED, (func_version/2, callable_p/4 --)) {
+            assert(PyFunction_Check(callable_p));
+            PyFunctionObject *func = (PyFunctionObject *)callable_p;
```

Comment:
What guarantee is that `func == frame->f_funcobj`?

Reply:
It's constant propagated. To be sure I added a runtime assert.